docs/package-migration-process-guide.md

```diff
@@ -0,0 +1,246 @@
+# Package Migration Process Guide
```

Comment:
More docs about migration that we can use for the design-tokens and eslint-plugin-design-tokens pacakge